TopMovies/Resources/Router.swift

```diff
@@ -43,9 +45,12 @@ class Router {
   private func setupCategoryVC(_ categoryId: MovieCategory.ID) -> MoviesCategoryVC {
     let categoryVc = MoviesCategoryVC()
     categoryVc.configureConnectionWith(
-      connector: CategoryVCConnector(categoryId: categoryId) { [unowned categoryVc] (props) in
-        categoryVc.connect(props: props)
-      })
+      connector: CategoryVCConnector(
+        categoryId: categoryId,
+        updateProps: { [unowned categoryVc] (props) in
+          categoryVc.connect(props: props)
+        },
```

Comment:
Is it possible to move that inside of the init of the `BaseConnector`?
Looks like a duplication. Moreover, that code is required to make connector work with vc.

Reply:
> Is it possible to move that inside of the init of the `BaseConnector`?
> Looks like a duplication. Moreover, that code is required to make connector work with vc.

How about Delegate pattern ?)